hathor/indexes/tips_index.py

```diff
@@ -61,3 +65,27 @@ def update_tx(self, tx: BaseTransaction, *, relax_assert: bool = False) -> None:
     @abstractmethod
     def __getitem__(self, index: float) -> Set[Interval]:
         raise NotImplementedError
+
+    def iter_mempool_tips_like(self, tx_storage: 'TransactionStorage') -> Iterator[Transaction]:
+        for interval in self[tx_storage.latest_timestamp + 1]:
+            tx = tx_storage.get_transaction(interval.data)
+            tx_meta = tx.get_metadata()
+            # XXX: skip blocks and tx-tips that have already been confirmed
+            if tx_meta.first_block is not None or tx.is_block:
+                continue
+            assert isinstance(tx, Transaction)
+            yield tx
+
+    def iter_mempool_like(self, tx_storage: 'TransactionStorage') -> Iterator[Transaction]:
+        from hathor.transaction.storage.traversal import BFSWalk
+
+        root = self.iter_mempool_tips_like(tx_storage)
```

Comment:
Can the root be an iterator?

Reply:
Yes. It explicitly accepts and works with `Union['BaseTransaction', Iterable['BaseTransaction']]`.

Reply:
Thanks!